### PR TITLE
Profile page

### PR DESCRIPTION
**Pull request by MrStashy**

### CODE REVIEW

**Review by PerikK — APPROVED** [review]

File: src/App.jsx [hunk 1]
```diff
@@ -1,54 +1,59 @@
...
+              />
+              <Route
+                path="welcome"
+                element={
+                  <ProtectedRoute disabledNav={true}>
+                    <Welcome />
+                  </ProtectedRoute>
+                }
+              />
+              <Route path="/search-results" element={<AllSearchResults />} />
```
Comment (vherus) [inline src/App.jsx:43]: I know this isn't yours, but shouldn't this be behind a protected route? If you think it should too, can you add it as a ticket to be worked on please?

File: src/components/UserDetails/index.jsx [hunk 2]
```diff
@@ -10,13 +10,15 @@ export default function UserDetails({ header = false }) {
...
+    
+    const stringToTitleCase = string => string.charAt(0).toUpperCase() + string.slice(1).toLowerCase();
 
     if (header) {        
         return (
             <div className="post-user-name">
                 <p>{userFullName}</p>
                 <small>
-                    {currentUser.role} {userCohort}
+                    {stringToTitleCase(currentUser.role)}{userCohort}
```
Comment (vherus) [inline src/components/UserDetails/index.jsx:21]: Should there be a space before injecting the userCohort?

File: src/pages/profile/index.jsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        result = "Bio";
+        break;
+      case "githubUsername":
+        result = "GitHub Username";
+        break;
+      default:
+        result = "Something went wrong";
+    }
+    return result;
+  };
```
Comment (vherus) [inline src/pages/profile/index.jsx:60]: I think an object map would be a little cleaner here:

```js
const labelMap = {
  cohortId: 'Cohort ID',
  role: 'Role',
  ...
}

...

label={labelMap[input]}
```

**Review by vherus — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 1 above)
Comment (PerikK) [reply src/App.jsx:43]: Sorry for popping up here. I did this. Why should it be in a protected route?

Comment (vherus) [reply src/App.jsx:43]: Should you be able to see search results, or the search results page, without being logged in?

Comment (PerikK) [reply src/App.jsx:43]: Hmmm. You can't see results, but you can get to the page. Should I fix it or or Will will?

Comment (vherus) [reply src/App.jsx:43]: I leave that up to your discretion 😄

File: src/components/UserDetails/index.jsx (hunk 2 above)
Comment (MrStashy) [reply src/components/UserDetails/index.jsx:21]: image
It renders correctly

File: src/pages/profile/index.jsx (hunk 3 above)
Comment (MrStashy) [reply src/pages/profile/index.jsx:60]: Great! Updated.

File: src/App.jsx (hunk 1 above)
Comment (MrStashy) [reply src/App.jsx:43]: This is a ticket now, here: https://github.com/boolean-uk/team-dev-client-c12/issues/142

**Review by vherus — APPROVED** [review]
LGTM

**Review by angustownsley — APPROVED** [review]

**Review by PerikK — APPROVED** [review]

**Review by homonoviscoding — APPROVED** [review]

**Pull request merged**